src/file.cpp

```diff
@@ -375,7 +375,12 @@ namespace libtorrent
 		{
 			std::string sep_path { path };
 			std::replace(sep_path.begin(), sep_path.end(), '/', '\\');
-			prepared_path = "\\\\?\\" + (is_complete(sep_path) ? sep_path : combine_path(current_working_directory(), sep_path));
+			if (!is_complete(sep_path))
+			{
+				std::string cwd_path { current_working_directory() };
+				sep_path = combine_path(cwd_path, sep_path);
+			}
+			prepared_path = "\\\\?\\" + sep_path;
```

Comment:
what do you think about a function like this then:

``string_view make_sv(std::string const& str) { return string_view(str); }`` ?

and say: ``combine_path(make_sv(current_working_directory()), sep_path)``